src/btree/bt_curnext.c

```diff
@@ -759,16 +748,26 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
             WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
         }
         cbt->page_deleted_count = 0;
-skip_page:
+
         if (F_ISSET(cbt, WT_CBT_READ_ONCE))
             LF_SET(WT_READ_WONT_NEED);
-        WT_ERR(__wt_tree_walk(session, &cbt->ref, flags));
+
+        /*
+         * If we are running with snapshot isolation, and not interested in returning tombstones, we
+         * could potentially skip pages. The skip function looks at the aggregated timestamp
+         * information to determine if something is visible on the page. If nothing is, the page is
+         * skipped.
+         */
+        if (session->txn->isolation == WT_ISO_SNAPSHOT &&
+          !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE))
```

Comment:
I'm confused by this -- the comment says "and not interested in returning tombstones", but the code is `not WT_CURSTD_IGNORE_TOMBSTONE`.

I'd read the code as "if the WT_CURSTD_IGNORE_TOMBSTONE flag is set, we want to ignore tombstones",  which doesn't match the comment.

This appears in two  places, of course.

Reply:
@keithbostic, the flag name, `WT_CURSTD_IGNORE_TOMBSTONE`, is the culprit. What this flags means is that, if set, caller is okay if the underlying cursor returns a value with globally visible tombstone. To the my best of knowledge, only history store cursor implementation applies this flag to the file cursor.

Maybe we can raise another ticket to think of a better name for this flag.

Reply:
@hbokhari, I would definitely support a ticket to change the flag name -- "ignore tombstone means tombstones are OK" isn't intuitive to me! :)

Reply:
Created WT-7915 to rename the flag to something more intuitive.